### ISSUE TITLE
Error: internal error: getTypeDescAux(tyEmpty)

### ISSUE THREAD

**Issue by akiradeveloper** [Araq/Nim#2632]
With v0.11.0, the xs of empty sequence within tuple can't be inferred the type and then raises error in title. Is this a compiler bug? This breaks my msgpack-nim library.

```nimrod
proc f(xs: seq[int]) =
  discard

proc g(t: tuple[n:int, xs:seq[int]]) =
  discard

when isMainModule:
  f(@[]) # OK
  g((1,@[1])) # OK
  g((0,@[])) # NG
```

**Comment by Varriount** [comment]
Can you give a revision that it broke at?

**Comment by akiradeveloper** [comment]
Of my msgpack-nim library? (https://github.com/akiradeveloper/msgpack-nim)

The latest 6e0d873f275af0469b868747946ba012ff111c2f is broken when we run `make quickcheck`. The task first generates random test cases and feeds them to msgpack library. So whether it fails or not depends on the randomness. I narrowed down the problem and which I showed above is the minimum case to reproduce the problem.

This problem didn't appear at Mar 25. So some degradation can be suspected. Since `f(@[])` is accepted by the compiler there is no way the third case `g((0, @[]))` isn't.
